ocaml-lsp-server/src/document_store.ml

```diff
@@ -14,3 +14,14 @@ let get store uri =
   | Some doc -> Ok doc
   | None ->
     Lsp.Import.Result.errorf "no document found with uri: %a" Lsp.Uri.pp uri
+
+let remove_document store uri =
```

Comment:
What about `Table.remove`?

Reply:
Isn't `Table.remove` an alias for `Hashtbl.remove`  which 
> removes the current binding of x in tbl, restoring the previous binding if it exists.

?

Reply:
Somehow github swallowed my comment.

This table shouldn't have more than one binding per key, so `Table.remove` is the right thing to use. You can just push that fix directly if you want.

Reply:
Are you sure ?
`Table.set` is an alias of `Table.add`, and uses `Lsp.Uri` (i.e. TextDocumentIdentifier) as a key instead of `VersionedTestDocumentIdentifier` (uri + version). It seems there is as many elements associated to an URI as there are modifications in the document since you opened the file.

I tested that by printing the size of the hashtable, and after closing a file it showed that the size of the table reduced by more than a hundred element after playing with a file for a few seconds. Maybe in that case we should use `Table.replace` instead

Reply:
I believe you may have found the memory leak that stdune introduced!

It's absolutely wrong to use `add` here and I thought that `set` was changed in `stdune` to be `replace`. Now you can fix 2 performance issues!

Reply:
Created the PR, merging it when the check passes :)